rnd/autogpt_libs/autogpt_libs/logging/config.py

```diff
@@ -0,0 +1,199 @@
+"""Logging module for Auto-GPT."""
+
+
+import logging
+import sys
+from pathlib import Path
+
+from pydantic import Field, field_validator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from .filters import BelowLevelFilter
+from .formatters import AGPTFormatter, StructuredLoggingFormatter
+
+LOG_DIR = Path(__file__).parent.parent.parent / "logs"
+LOG_FILE = "activity.log"
+DEBUG_LOG_FILE = "debug.log"
+ERROR_LOG_FILE = "error.log"
+
+SIMPLE_LOG_FORMAT = "%(asctime)s %(levelname)s  %(title)s%(message)s"
+
+DEBUG_LOG_FORMAT = (
+    "%(asctime)s %(levelname)s %(filename)s:%(lineno)d" "  %(title)s%(message)s"
+)
+
+
+class LoggingConfig(BaseSettings):
+
+    level: str = Field(
+        default="INFO",
+        description="Logging level",
+        validation_alias="LOG_LEVEL",
+    )
+
+    enable_cloud_logging: bool = Field(
+        default=False,
+        description="Enable logging to Google Cloud Logging",
+        validation_alias="ENABLE_CLOUD_LOGGING",
+    )
+
+    enable_file_logging: bool = Field(
+        default=True,
+        description="Enable logging to file",
+        validation_alias="ENABLE_FILE_LOGGING",
+    )
+
+    enable_console_logging: bool = Field(
+        default=True,
+        description="Enable logging to console",
+        validation_alias="ENABLE_CONSOLE_LOGGING",
+    )
+
+    enable_json_logging: bool = Field(
+        default=False,
+        description="Enable logging to JSON file",
+        validation_alias="ENABLE_JSON_LOGGING",
+    )
+
+    # File output
+    log_dir: Path = Field(
+        default=LOG_DIR,
+        description="Log directory",
+    )
+
+    model_config = SettingsConfigDict(
+        env_prefix="",
+        env_file=".env",
+        env_file_encoding="utf-8",
+        extra="ignore",
+    )
+
+    @field_validator("level", mode="before")
+    @classmethod
+    def parse_log_level(cls, v):
+        if isinstance(v, str):
+            v = v.upper()
+            if v not in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]:
+                raise ValueError(f"Invalid log level: {v}")
+            return v
+        return v
+
+
+def configure_logging(
+        force_cloud_logging: bool = False
```

Comment:
1. why keep this single parameter?
2. double indent, is lint broken?

Reply:
Because it is used in the autogpt_server the rest where unused